### ISSUE TITLE
[FR] Quarterly file support

### ISSUE THREAD

**Issue by zalegrala** [renerocksai/telekasten.nvim#90]
**Please confirm**
- [x] I am running the latest version of this plugin

**Is your feature request related to a problem? Please describe.**

Some organizations use a goal-setting process called OKRs, where once a quarter you set some goals.  It is now that time where I fill out mine and submit them up the chain, but in thinking about and using telekasten for the last few weeks, it occurs to me that much of this could be wrapped into this plugin if quartelry support were added.

**Describe the solution you'd like**
In the same way that we have a weekly template, and a weekly file, I think having a quarterly file and template would be pretty great.  Template variables like `{{thisquarter}}` or `{{lastquarter}}` or some such could also be handy.  A weekly template could have a link to this quarters goals to make it easier to refer back, etc.

**Describe alternatives you've considered**
Right now I'm using a tag to go back to a file that I know has what I want, which is just a weekly note.

**Additional context**
Quarters are not always aligned to the year, so having a way to tell when the quarter starts would be handy, but perhaps tricky to implement.

**Comment by zalegrala** [comment]
In thinking about this a little more, I think it would be sufficient to have a template variable that would be able to contain the quarter.  Perhaps there could be a config variable to say when to anchor the quarter beginning.

**Comment by lambtho12** [comment]
> it would be sufficient to have a template variable that would be able to contain the quarter

Agree. I do not think it is necessary to have a specific file for each possible subdivision. People will want semesters, bi-annual, bi-weekly, whatever else and we will never see the end of it.

What we could do instead is creating a `goto_specialDate()` that would always start by prompting the user pick a template (monthly, quarterly, whateverly) and save them in `journal/special/` with a custom filename like `S2022-<your_input>.md`. The custom filename with a fixed part would help telekasten determine if it is a regular note or a Special Date note. In addition to that, some simple templates for quarters, months, etc could be added.

For changing the Q1 reference date, I do not exactly know how we could do it. Ideally that would happen within the template itself. So if you make quarterly notes for multiple companies, you can align each note on the appropriate calendar. Maybe adding a `{{quarterstart=DD-MM}}` somewhere in the template would be the easiest solution.

I also feel that if we go that route, we should refactor a bit and put the templates variables and the calculation of dates in a separate file so it becomes easier to maintain.

**Comment by zalegrala** [comment]
The more I use this plugin, the less I think this is specifically required.  I like the discussion on the Custom Date Format.  What I've done is just put the specific quarter I'm referencing as a link in the template, and then once a quarter I will change one number.  That seems reasonable enough to me.  We can leave this open if you wish, but the other conversationin https://github.com/renerocksai/telekasten.nvim/discussions/101 with a more general approach to flexibility seems like a better approach to me.  Consider this a mostly non-issue for me at the moment.  Thanks for the discussion.
